bootstrap.sh

```diff
@@ -0,0 +1,234 @@
+	    sudo /usr/sbin/sysadminctl -fullName "Nix build user $i" \
+		 -home /var/empty \
+		 -shell /sbin/nologin \
+		 -UID $(expr 30000 + $i) \
+		 -addUser nixbld$i >&2 \
+		|| exit_message "Problem creating user: nixbld${i}"
+
+	    sudo dscl . -create /Users/nixbld$i IsHidden 1 || \
+		exit_message "Problem setting 'IsHidden' for user: nixbld${i}"
+	    sudo dscl . -create /Users/nixbld$i UserShell /sbin/nologin || \
+		exit_message "Problem setting shell for user: nixbld${i}"
+	    sudo /usr/sbin/dseditgroup -o edit -t user -a nixbld$i nixbld || \
+		exit_message "Problem setting primary group for user: nixbld${i}"
+	    sudo /usr/bin/dscl . -create /Users/nixbld$i PrimaryGroupID 30000 >&2 || \
+		exit_message "Problem setting PrimaryGroupID for user: nixbld${i}"
+	else
+	    echo -e "It looks like the "$YELLOW"nixbld${i}"$ESC" user already exists!"
+	fi
+    done
+
+    [ ! -z $CREATE_DAEMON_USERS ] && exit 0
+}
+
+# Installer
+install(){
+    echo -e ""$BLUE_UL"Welcome to the nix-darwin installer!"$ESC""
+
+    # Prompt for nix package upgrade
+    echo -e "Ensuring "$YELLOW"nixpkgs"$ESC" version meets requirements..."
+    echo -e "To do this, the following will be run to upgrade the "$YELLOW"nix"$ESC" package:"
+    echo -e ""$YELLOW"nix-env -iA nixpkgs.nix"$ESC"\n"
+    echo "If you have a recent install of Nix, this may not be necessary, but should not cause any harm to run"
+    while true ; do
+	read -p "Would you like to upgrade? [y/n] " ANSWER
+	case $ANSWER in
+	    y|Y)
+		echo "Proceeding with upgrade..."
+		nix-env -iA nixpkgs.nix
+		break
+		;;
+	    n|N)
+		echo "Proceeding without upgrade..."
+		break
+		;;
+	    *)
+		echo "Please answer 'y' or 'n'..."
+	esac
+    done
+
+    sudo_prompt
+
+    # Link run directory
+    echo "Setting up /run..."
+    test -L /run || sudo ln -snf private/var/run /run
+
+    # Fetch the nix-darwin repo
+    echo -e ""$YELLOW"Fetching nix-darwin repo..."$ESC""
+    nix-env -p "/nix/var/nix/profiles/per-user/$USER/darwin" \
+    	    --set $(nix-prefetch-url --unpack --print-path \
+    				     --name nix-darwin-17.03 \
+    				     https://github.com/LnL7/nix-darwin/archive/master.tar.gz \
+    			| grep nix-darwin)
+    ln -sfn "/nix/var/nix/profiles/per-user/$USER/darwin" "$HOME/.nix-defexpr/darwin"
+
+    # Copy the example configuration
+    echo -e "Copying example configuration to "$YELLOW"~/.nixpkgs/darwin-configuration.nix"$ESC"..."
+
+    mkdir -p "$HOME/.nixpkgs"
+    cp "$HOME/.nix-defexpr/darwin/modules/examples/simple.nix" "$HOME/.nixpkgs/darwin-configuration.nix"
+
+    # Bootstrap build using default nix.nixPath
+    echo "Bootstrapping..."
+    export NIX_PATH=darwin=$HOME/.nix-defexpr/darwin:darwin-config=$HOME/.nixpkgs/darwin-configuration.nix:$NIX_PATH
+    $(nix-build '<darwin>' -A system --no-out-link)/sw/bin/darwin-rebuild build
+    $(nix-build '<darwin>' -A system --no-out-link)/sw/bin/darwin-rebuild switch
+
+    # Source generated bashrc
+    . /etc/static/bashrc
```

Comment:
This was causing problems in combination with `set -e`, I disabled completion for the example.